spec/requests/custom_buttons_spec.rb

```diff
@@ -0,0 +1,206 @@
+    end
+
+    it 'lists all custom buttons with an appropriate role' do
+      api_basic_authorize collection_action_identifier(:custom_buttons, :read, :get)
+      cb_href = api_custom_button_url(nil, cb)
+
+      get(api_custom_buttons_url)
+
+      expected = {
+        'count'     => 1,
+        'subcount'  => 1,
+        'name'      => 'custom_buttons',
+        'resources' => [
+          hash_including('href' => cb_href)
+        ]
+      }
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to include(expected)
+    end
+  end
+
+  describe 'GET /api/custom_buttons/:id' do
+    it 'does not let you query custom buttons without an appropriate role' do
+      api_basic_authorize
+
+      get(api_custom_button_url(nil, cb))
+
+      expect(response).to have_http_status(:forbidden)
+    end
+
+    it 'can query a custom button by its id' do
+      api_basic_authorize action_identifier(:custom_buttons, :read, :resource_actions, :get)
+
+      get(api_custom_button_url(nil, cb))
+
+      expected = {
+        'id'   => cb.id.to_s,
+        'name' => "custom_button"
+      }
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to include(expected)
+    end
+  end
+
+  describe 'POST /api/custom_buttons' do
+    it 'can create a new custom button' do
+      api_basic_authorize collection_action_identifier(:custom_buttons, :create)
+
+      cb_rec = {
+        'name'             => 'Generic Object Custom Button',
+        'description'      => 'Generic Object Custom Button description',
+        'applies_to_class' => 'GenericObjectDefinition',
+        'options'          => {
+          'button_icon'  => 'ff ff-view-expanded',
+          'button_color' => '#4727ff',
+          'display'      => true,
+        },
+      }
+      post(api_custom_buttons_url, :params => cb_rec)
+
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body['results'].first).to include(cb_rec)
+    end
+
+    it 'can edit custom buttons by id' do
+      api_basic_authorize collection_action_identifier(:custom_buttons, :edit)
+
+      request = {
+        'action'    => 'edit',
+        'resources' => [
+          { 'id' => cb.id.to_s, 'name' => 'updated 1' },
+        ]
+      }
+      post(api_custom_buttons_url, :params => request)
+
+      expected = {
+        'results' => a_collection_including(
+          a_hash_including('id' => cb.id.to_s, 'name' => 'updated 1'),
+        )
+      }
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to include(expected)
+    end
+  end
+
+  describe 'POST /api/custom_buttons/:id' do
+    it 'can update a custom buttons by id' do
+      api_basic_authorize action_identifier(:custom_buttons, :edit)
+
+      request = {
+        'action'      => 'edit',
+        'name'        => 'Generic Object Custom Button Updated',
+        'description' => 'Generic Object Custom Button description Updated',
```

Comment:
Similarly if you can edit the resource action details here too that would be great 😁

Reply:
in 5104806